File: src/Cpp/examples/example03_runtime_pipeline.cpp
#include <iostream>
#include <string>

#include "pipeline_services/core/runtime_pipeline.hpp"
#include "pipeline_services/examples/text_steps.hpp"

int main() {
  pipeline_services::core::RuntimePipeline<std::string> runtime_pipeline(
    "example03_runtime_pipeline",
    false,
    "  Hello   Runtime  "
  );
  runtime_pipeline.add_action(pipeline_services::examples::strip);
  runtime_pipeline.add_action(pipeline_services::examples::normalize_whitespace);
  std::cout << "runtimeValue=" << runtime_pipeline.value() << std::endl;

  const auto frozen_pipeline = runtime_pipeline.freeze();
  const auto result = frozen_pipeline.execute("  Hello   Frozen  ");
  std::cout << "frozenValue=" << result.context << std::endl;
  return 0;
}

File: src/Rust/examples/example03_runtime_pipeline.rs
use pipeline_services::examples::text_steps::{normalize_whitespace, strip};
use pipeline_services::RuntimePipeline;

fn main() {
  let mut runtime_pipeline = RuntimePipeline::new("example03_runtime_pipeline", false, "  Hello   Runtime  ".to_string());
  runtime_pipeline.add_action(strip);
  runtime_pipeline.add_action(normalize_whitespace);
  match runtime_pipeline.value() {
    Some(value) => println!("runtimeValue={value}"),
    None => println!("runtimeValue=<none>"),
  }

  let frozen_pipeline = runtime_pipeline.freeze();
  let result = frozen_pipeline.run("  Hello   Frozen  ".to_string());
  println!("frozenValue={}", result.context);
}
